### ISSUE TITLE
cudagraph and stream capture issue

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos#2606]
When creating a cudagraph via stream capture, where the kernel to be captured is the first `parallel_*` call, results in testing failures like the following:

```
terminate called after throwing an instance of 'std::runtime_error'
  what():  cudaMemcpyToSymbol( Kokkos::Impl::g_device_cuda_lock_arrays , & Kokkos::Impl::g_host_cuda_lock_arrays , sizeof(Kokkos::Impl::CudaLockArrays) ) error( cudaErrorStreamCaptureImplicit): operation would make the legacy stream depend on a capturing blocking stream /ascldap/users/ndellin/kokkos-kernels/testing/TriSolveCudaOMP-pr/kokkos/install/include/Cuda/Kokkos_Cuda_KernelLaunch.hpp:436
Traceback functionality not available
```

I can workaround this by including a dummy `parallel_for` call that does nothing prior to the stream capture.

I spoke with @crtrott about this, he said this may not occur when rdc is enabled, and the issue is that the first call to a `parallel_*` within each object file will result in initializations that can't be captured by the stream capture (like the lock arrays in this case).

I don't have a simple code snip but will have a kokkos-kernels PR submitted later that I'll add reference to.

It wasn't clear to me if the guidance for using cuda graphs with Kokkos will be to require rdc or if some enhancement will occur to support this. I don't think this is bug, but either enhancement or documentation (use rdc with cuda graphs).

**Comment by ndellingwood** [comment]
Adding @crtrott, should this be marked documentation regarding stream capture usage with Kokkos when rdc is available vs unavailable?

**Comment by PhilMiller** [comment]
@crtrott This seems like the sort of thing where many relevant conditions may have changed in the time since it was opened. Disposition?

**Comment by crtrott** [comment]
Yeah we probably should check if this is still an issue. I think we might have fixed this.
